source/particle/generator/interface.cc

```diff
@@ -136,14 +137,101 @@ namespace aspect
 
       template <int dim>
       void
-      Interface<dim>::declare_parameters (ParameterHandler &)
-      {}
+      Interface<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection ("Postprocess");
+        {
+          prm.enter_subsection ("Tracers");
+          {
+            prm.enter_subsection("Generator");
+            {
+              prm.declare_entry("Generate particles multiple times", "false",
+                                Patterns::Bool(),
+                                "Set to generate particles at every time step number interval specified.");
+              prm.declare_entry("Time step number interval", "100",
+                                Patterns::Integer(),
+                                "Generate particles at ever Time step number interval");
+              prm.declare_entry("Initially generate particles", "true",
+                                Patterns::Bool(),
+                                "Set to false to prevent the geneartion of particles at time step number 0.");
+            }
+            prm.leave_subsection ();
+          }
+          prm.leave_subsection ();
+        }
+        prm.leave_subsection ();
+      }
 
       template <int dim>
       void
       Interface<dim>::parse_parameters (ParameterHandler &)
       {}
 
+      template <int dim>
+      void
+      Interface<dim>::parse_global_parameters (ParameterHandler &prm)
+      {
+        std::string name;
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            name = prm.get ("Particle generator name");
+            prm.enter_subsection("Generator");
+            {
+              repeatedlyGenerateParticles = prm.get_bool("Generate particles multiple times");
+              if (repeatedlyGenerateParticles)
+                nTimeStepNumberInterval = prm.get_integer("Time step number interval");
+              generateParticlesInitially = prm.get_bool("Initially generate particles");
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+
+        if (repeatedlyGenerateParticles)
+          {
+            AssertThrow((name != "probability density function") && (name != "random uniform"),
```

Comment:
You should not rely on which plugins are implemented right now. If someone adds a plugin, but forgets to add it to this list your check will not work anymore. Better check if one of the supported plugins is selected.